x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/UnsignedLongFieldMapper.java

```diff
@@ -705,4 +707,29 @@ public void doValidate(MappingLookup lookup) {
             );
         }
     }
+
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        if (hasDocValues == false) {
```

Comment:
The `unsigned_long` field type can be used a dimension / metric in tsdb indices.
Adding synthetic source support for this, since that is the only supported source mode for tsdb indices.

Reply:
:+1: I imagine there are a bunch of tests that need poking for this.